File: POC_Tasks/PathBetweenTwoPoints/Php/src/php/PathBetweenTwoPointsPedestrian.php
<?php


class PathBetweenTwoPointsPedestrian
{
    function __construct($key='choisirgeoportail', $origin='1.4296871423721542,48.9491452350537', $destination='1.8691402673721758,48.80463579895799', $url='http://wxs.ign.fr/KEY/itineraire/rest/route.json?') // constructor
    {
        $this->key = $key; // load key
        $this->origin = $origin; // load origin point
        $this->destination = $destination; // load destination point
        $this->url = $url; // load url
    }

    function build_url_using_key() // build the url
    {
        return str_replace("KEY", $this->key, $this->url);  # add the key on url
    }

    function build_data_options() // build the data
    {
        $data = array(
            'origin' => $this->origin,
            'destination' => $this->destination,
            'method' => 'DISTANCE',
            'graphName' => 'Pieton'
        ); // give value on each option

        return http_build_query($data, '', '&'); // build data string
    }

    function get_http_result()
    {

        $curl_handle=curl_init(); // check extension on php.ini
        curl_setopt($curl_handle, CURLOPT_URL,$this->build_url_using_key() . $this->build_data_options());
        curl_setopt($curl_handle, CURLOPT_CONNECTTIMEOUT, 2);
        curl_setopt($curl_handle, CURLOPT_RETURNTRANSFER, 1);
        curl_setopt($curl_handle, CURLOPT_USERAGENT, 'ItinerairePieton');
        $query = curl_exec($curl_handle);
        curl_close($curl_handle);

        return $query;
    }
}

File: POC_Tasks/PathBetweenTwoPoints/Python/pathbetweentwopointspedestrian.py
import requests

key = "choisirgeoportail"  # the key is choisirgeoportail for test and development
lat_start, lon_start = "1.4296871423721542", "48.9491452350537"  # the start point
lat_end, lon_end = "1.8691402673721758", "48.80463579895799"  # the end point

get_request = "http://wxs.ign.fr/[KEY]/itineraire/rest/route.json?origin=[lat_start],[lon_start]&destination=[lat_end],[lon_end]&&method=DISTANCE&graphName=Pieton"  # the REST api request
get_request = get_request.replace("[KEY]", "%s").replace("[lat_start]", "%s").replace("[lat_end]", "%s").replace("[lon_start]", "%s").replace("[lon_end]", "%s")  # format the string
get_request = get_request % (key, lat_start, lon_start, lat_end, lon_end)  # add parameters to the request

r = requests.get(get_request)  # execute the request and get the result
print(r.text)  # print the request result
